File: AdventOfCode2021/Day1/Day1b.fs
module Day1b

open System.IO

let solver =
    let measurements = File.ReadLines "Inputs\\day1.txt" |> Seq.map int |> Seq.toList
    measurements[3..]
        |> List.mapi (fun i _ -> 
            if ((measurements[i + 1] + measurements[i + 2] + measurements[i + 3]) > (measurements[i] + measurements[i + 1] + measurements[i + 2]))
                then 1 else 0
                )
        |> List.sum
        |> printfn "%d"

File: AdventOfCode2024/Day1/Day1B.cs
using System.Collections.Immutable;

namespace AdventOfCode2024.Day1;

public class Day1B : ISolver
{
    public void Solve()
    {
        var list1 = new List<int>();
        var list2 = new List<int>();
        
        var lines = File.ReadLines("Inputs\\day1.txt");
        foreach (var line in lines)
        {
            var splitLine = line.Split("   ");
            list1.Add(int.Parse(splitLine[0]));
            list2.Add(int.Parse(splitLine[1]));
        }

        var counts = list2.GroupBy(i => i).ToImmutableDictionary(group => group.Key, group => group.Count());

        var similarity = list1.Sum(i => i * counts.GetValueOrDefault(i, 0));

        Console.WriteLine(similarity);
    }
}
